src/components/sbb-checkbox/sbb-checkbox.tsx

```diff
@@ -0,0 +1,132 @@
+import { Component, Prop, h, Event, EventEmitter, Watch } from '@stencil/core';
+
+let nextId = 0;
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-checkbox.scss',
+  tag: 'sbb-checkbox',
+})
+export class SbbCheckbox {
+  private _checkbox: HTMLInputElement;
+
+  /** Whether the checkbox is checked. */
+  @Prop({ mutable: true }) public checked: boolean;
+
+  /** Value of checkbox. */
+  @Prop() public value?: string;
+
+  /** Name of the checkbox */
+  @Prop() public name?: string;
+
+  /** Id of the internal input element - default id will be set automatically. */
+  @Prop() public inputId = `sbb-checkbox-${++nextId}`;
+
+  //** the svg for the true state */
+  @Prop() public labelIcon: '';
+
+  /** The disabled prop for the disabled state. */
+  @Prop() public disabled!: boolean;
+
+  /** The required prop for the required state. */
+  @Prop() public required?: boolean;
+
+  /** Whether the checkbox is tristated. */
+  @Prop() public tristated?: boolean;
+
+  /** The label position relative to the labelIcon. Defaults to false */
+  @Prop() public labelReversed = false;
+
+  /** Whether the checkbox label has spacing to the labelIcon. */
+  @Prop() public labelSpace = false;
+
+  /** The aria-label prop for the hidden input. */
+  @Prop() public acceccibilityLabel?: string;
+
+  /** The aria-labelledby prop for the hidden input. */
+  @Prop() public acceccibilityLabelledby?: string;
+
+  /** The aria-describedby prop for the hidden input. */
+  @Prop() public acceccibilityDescribedBy?: string;
+
+  /** Event for emiting whenever selection is changed. */
+  @Event() public sbbCheckboxChange: EventEmitter;
+
+  /** render the svg acording to the state */
+  private _renderStateIcon(): JSX.Element {
+    if (this.checked === true) {
+      return (
+        <svg
+          width="10"
+          height="8"
+          viewBox="0 0 10 8"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <path
+            d="M1 4.33 3.462 7 9 1"
+            stroke="#EB0000"
+            stroke-width="2"
+            stroke-linecap="round"
+            stroke-linejoin="round"
+          />
+        </svg>
+      );
+    } else if (this.tristated === true && this.checked === false) {
+      return (
+        <svg xmlns="http://www.w3.org/2000/svg" width="8" height="2" viewBox="0 0 8 2" fill="none">
+          <path
+            d="M1 1H7"
+            stroke="#EB0000"
+            stroke-width="2"
+            stroke-linecap="round"
+            stroke-linejoin="round"
+          />
+        </svg>
+      );
+    }
+  }
+
+  @Watch('checked')
```

Comment:
Not sure if this is stencil specific, but I seems wrong to fire an changeEvent when the change comes from the parent component (passed) (could also lead to infinite update loops). Consider fireing an update event from the input comp directly and don't "maintain" the checked state in the component directly. The comp should be stateless.

Reply:
https://github.com/lyne-design-system/lyne-components/pull/1283#discussion_r916842784

can you check Lukas comment on this subject?

Reply:
resolved: props are not reactive -> watch is needed